### PR TITLE
Make modules static libraries

### PR DESCRIPTION
**Pull request by neithernut**
This PR reorganized the build environmane,t which brings many advantages, e.g. easier creation of tests and faster incremental builds.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Most of this PR seems to be scripted. I did a short review of the most stuff, but I'd like @TheNeikos to do a review as well.

**Comment by neithernut** [comment]
The creation or some commits was scripted. The dependency deduction was performed by hand.

**Comment by neithernut** [comment]
May this merged already?

**Comment by matthiasbeyer** [comment]
The latest commit has a typo in the commit message.

I'd also suggest that you rebase on the latest develop, to be sure there are no fuckups around...

**Comment by neithernut** [comment]
This should be checked by the CI, as we perform a merge just before comilation and testing now, but i'll do it nonetheless.

---

Done.

**Comment by matthiasbeyer** [comment]
The problem is that the build doesn't get triggered when develop updates, so there is still a way to crash the whole thing.

But well... will review now (again).

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -26,44 +26,16 @@ include_directories(
...
-    ${WAYLAND_CURSOR_INCLUDE_DIRS}
-    ${WAYLAND_SERVER_INCLUDE_DIRS}
-    ${XKB_COMMON_INCLUDE_DIRS}
-    ${YAJL_INCLUDE_DIRS}
-    ${UUID_INCLUDE_DIRS}
 )
 
 #
-# Add definitions
+# Set devintions
```
Comment (matthiasbeyer) [inline src/CMakeLists.txt]: Typo.

**Comment by matthiasbeyer** [comment]
Please fix the one typo, besides: ACK.

File: src/CMakeLists.txt (hunk 1 above)
Comment (neithernut) [inline src/CMakeLists.txt]: I wonder how this happened...

Comment (matthiasbeyer) [inline src/CMakeLists.txt]: Me two!

**Comment by neithernut** [comment]
Note that you have to re-run CMake/clean the cache when transitting to the new setup. I'd suggest blindly rebasing _every_ PR.

**Comment by matthiasbeyer** [comment]
I rebased everything I have.

**Pull request merged**